### PR TITLE
[wp-env] Allow DB connection from any host

### PR DESCRIPTION
**Pull request by iazel**
## What?
Fix `wp-env` utility so that it can work with newer version of MariaDB docker image. Related issue: #25717

## Why?

As explained in [MySQL docs](https://dev.mysql.com/doc/en/docker-mysql-more-topics.html#docker_var_mysql-root-host), by default the root user can be accessed only from inside the container and any other access is denied.

Given that we are only interested in development, what we really want is to allow connection from any host.

## How?
Set the `MYSQL_ROOT_HOST` env variable for MariaDB containers to allow access from any host.

### CODE REVIEW

**Comment by iazel** [comment]
I'm pretty sure tests failure is unrelated to this change. Is there any way to just try them again?

**Review by alexstine — APPROVED** [review]
LGTM.

**Comment by alexstine** [comment]
@iazel The tests have been picky this week. I'll re-run.

Thanks.

**Comment by alexstine** [comment]
Okay, finally, tests pass. Let me give this fresh test later today and I'll merge if all is good.

**Comment by iazel** [comment]
> Okay, finally, tests pass. Let me give this fresh test later today and I'll merge if all is good.

Ok, thank you for taking care of this

**Comment by alexstine** [comment]
Tested this locally and all seems fine.

@noahtallen Can you give this a quick review? Saw you were marked as a code owner.

**Review by noahtallen — APPROVED** [review]
Looks good!

**Comment by noahtallen** [comment]
I agree that this doesn't need to be hardened just for the development environment.

**Pull request merged**
